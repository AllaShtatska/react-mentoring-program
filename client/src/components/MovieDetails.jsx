import '../static/inline.css';
import '../static/margin-left.css';

import React from 'react';

class MovieDetails extends React.Component {
    render(){
        return(
            <div>
                <div className='inline'><img src={this.props.selectedMovie.poster_path}/></div>
                <div className="inline">
                    <div className="inline">{this.props.selectedMovie.title}</div>
                    <div className="inline margin-left">{this.props.selectedMovie.vote_average}</div>
                    <div>{this.props.selectedMovie.genres}</div>
                    <div className="inline">{this.props.selectedMovie.release_date}</div>
                    <div className="inline margin-left">{this.props.selectedMovie.runtime} min</div>
                    <div>{this.props.selectedMovie.overview}</div>
                </div>
            </div>
        );
    }
}

export default MovieDetails;