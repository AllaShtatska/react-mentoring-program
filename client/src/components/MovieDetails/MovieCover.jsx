import '../../static/image';

import React from 'react';

const MovieCover=({movie}) => (
    <div className='inline'><img src={movie.poster_path}/></div>
);

export default MovieCover;