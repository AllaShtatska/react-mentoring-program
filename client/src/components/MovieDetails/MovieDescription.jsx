import React from 'react';

const MovieDescription=({movie}) => (
    <div>{movie.overview}</div>
);

export default MovieDescription;