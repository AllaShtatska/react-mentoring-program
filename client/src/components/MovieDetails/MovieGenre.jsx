import React from 'react';

const MovieGenre=({movie}) => (
    <div>{movie.genre}</div>
);

export default MovieGenre;