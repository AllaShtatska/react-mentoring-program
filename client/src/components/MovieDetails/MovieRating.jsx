import '../../static/inline.css';
import '../../static/margin-left.css';
import React from 'react';

const MovieRating=({movie}) => (
    <div className="inline margin-left">{movie.vote_average}</div>
);

export default MovieRating;