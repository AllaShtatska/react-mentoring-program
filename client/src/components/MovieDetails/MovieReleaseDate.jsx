import '../../static/inline';

import React from 'react';

const MovieReleaseDate=({movie}) => (
    <div className="inline">{movie.release_date}</div>
);

export default MovieReleaseDate;