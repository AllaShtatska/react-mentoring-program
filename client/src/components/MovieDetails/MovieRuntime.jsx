import '../../static/inline.css';
import '../../static/margin-left.css';
import React from 'react';

const MovieRuntime=({movie}) => (
    <div className="inline margin-left">{movie.runtime} min</div>
);

export default MovieRuntime;