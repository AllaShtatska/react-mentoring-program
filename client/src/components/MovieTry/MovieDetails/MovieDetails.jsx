import React from 'react';
import MovieCover from './MovieCover';
import MovieTitle from './MovieTitle'
import MovieRating from './MovieRating';
import MovieGenre from "./MovieGenre";
import MovieReleaseDate from './MovieReleaseDate'
import MovieRuntime from './MovieRuntime'
import MovieDescription from './MovieDescription'

class MovieDetails extends React.Component {
    render(){
        return(
            <div>
                <MovieCover movie={this.props.selectedMovie}/>
                <div className="inline">
                    <MovieTitle movie={this.props.selectedMovie}/>
                    <MovieRating movie={this.props.selectedMovie}/>
                    <MovieGenre movie={this.props.selectedMovie}/>
                    <MovieReleaseDate movie = {this.props.selectedMovie}/>
                    <MovieRuntime movie = {this.props.selectedMovie}/>
                    <MovieDescription movie={this.props.selectedMovie}/>
                </div>
            </div>
        );
    }
}

export default MovieDetails;