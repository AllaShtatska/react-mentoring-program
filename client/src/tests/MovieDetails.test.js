import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetails from '../components/MovieDetails';

it ('renders correctly', ()=>{
    const movie = {
        id: "1",
        title: "Kill Bill: Vol1",
        genres: ["Actions&Advanture"],
        release_date: "October 10, 2003",
        vote_average: 10,
        poster_path: '../images/Kill_Bill_Volume_1.png',
        runtime: 111,
        overview: "Kill Bill: Vol1 overview"
    };

    const component = renderer.create(<MovieDetails selectedMovie={movie}/>);

    var jsonComponent = component.toJSON();
    expect(jsonComponent).toMatchSnapshot();
});