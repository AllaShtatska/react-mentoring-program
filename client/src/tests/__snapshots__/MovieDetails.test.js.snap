// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="inline"
  >
    <img
      src="../images/Kill_Bill_Volume_1.png"
    />
  </div>
  <div
    className="inline"
  >
    <div
      className="inline"
    >
      Kill Bill: Vol1
    </div>
    <div
      className="inline margin-left"
    >
      10
    </div>
    <div>
      Actions&Advanture
    </div>
    <div
      className="inline"
    >
      October 10, 2003
    </div>
    <div
      className="inline margin-left"
    >
      111
       min
    </div>
    <div>
      Kill Bill: Vol1 overview
    </div>
  </div>
</div>
`;
